src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+--!strict
+
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local MarketplaceService = game:GetService("MarketplaceService")
+
+local Packages = ReplicatedStorage.Packages
+local Knit = require(Packages.Knit)
+local GameAnalytics = require(Packages.GameAnalytics)
+
+local CURRENCIES = { "Coins" } -- List of all in-game currencies
+local BUILD = "0.0.1" -- Game version
+local GAME_KEY = "xxxxxxxxxxxxxxxx"
+local SECRET_KEY = "xxxxxxxxxxxxxxxx"
+local LOG_DEV_PRODUCT_PURCHASES = true
+local RESOURCE_EVENT_TYPES = { -- (example: player gained coins in a mission is a
+	-- "Reward" event type, player purchasing coins with Robux is a "Purchase" event type)
+	"Reward",
+	"Purchase",
+	"Shop",
+	"Loot",
+	"Combat"
+}
+local GAMEPASS_IDS = {} -- List of all gamepass ids in the game
+local CUSTOM_DIMENSIONS = {
+	-- Uncomment and fill each dimension as needed
+	-- Refer to https://docs.gameanalytics.com/advanced-tracking/custom-dimensions
+	-- for more information about dimensions
+	
+	-- DIMENSION_01 = {};
+	-- DIMENSION_02 = {};
+	-- DIMENSION_03 = {};
+}
+
```

Comment:
The less developers have to edit the source-code behind the boilerplate, the easier it'll be to maintain/keep consistent standards/behavior.

Consider adding a function that lets variables such as BUILD, GAME_KEY, SECRET_KEY, CURRENCIES, and other static variables be set before knit start is fired - that way developers don't need to edit the source code to this service directly to make changes. 

Make sure the placeholder variables have their types defined (Eg. GAME_KEY: string, SECRET_KEY: string, LOG_DEV_PRODUCT_PURCHASES: boolean etc.)